src/hotspot/share/opto/doCall.cpp

```diff
@@ -339,17 +338,24 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
     // number of implementors for decl_interface is 0 or 1. If
     // it's 0 then no class implements decl_interface and there's
     // no point in inlining.
-    if (call_does_dispatch && bytecode == Bytecodes::_invokeinterface) {
-      ciInstanceKlass* declared_interface =
-          caller->get_declared_method_holder_at_bci(bci)->as_instance_klass();
+    if (call_does_dispatch && is_interface) {
+      ciInstanceKlass* declared_interface = nullptr;
+      if (orig_callee->intrinsic_id() == vmIntrinsics::_linkToInterface) {
+        // MemberName doesn't keep symbolic information once resolution is over, but
+        // resolved method holder can be used as a conservative approximation.
```

Comment:
Is "symbolic information" the referenced interface and the "resolved method holder" the declaring interface? I think including "referenced" vs "declared" would be more clear.